src/main/java/com/google/devtools/build/lib/hash/AbstractHashFunction.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2017 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.devtools.build.lib.hash;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkPositionIndexes;
+
+import com.google.common.hash.Funnel;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import com.google.errorprone.annotations.Immutable;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+/**
+ * Skeleton implementation of {@link HashFunction} in terms of {@link #newHasher()}.
+ *
+ * <p>This is AbstractHashFunction.java from Guava (which is not yet open sourced). See
+ * https://github.com/google/guava/issues/938.
+ *
+ * <p>TODO(lowasser): make public
```

Comment:
Update the documentation a bit to refer to guava instead of duplicating?

Reply:
(Removed this file)